AndroidSDKCore/src/main/java/com/leanplum/Leanplum.java

```diff
@@ -1041,23 +1041,26 @@ private static void resumeInternal() {
    */
   private static void startHeartbeat() {
     synchronized (heartbeatLock) {
-      heartbeatExecutor = Executors.newSingleThreadScheduledExecutor();
-      heartbeatExecutor.scheduleAtFixedRate(new Runnable() {
-        public void run() {
-          try {
-            Request.post(Constants.Methods.HEARTBEAT, null).sendIfDelayed();
-          } catch (Throwable t) {
-            Util.handleException(t);
+      if (heartbeatExecutor == null) {
+        heartbeatExecutor = Executors.newSingleThreadScheduledExecutor();
```

Comment:
This is getting a little tabbed out to unreadability. Can we extract a `createHeartbeat` function to call `if (heartbeatExecutor == null)` ?